packages/oauth/oauth-provider/src/types/email.ts

```diff
@@ -0,0 +1,17 @@
+import { isEmailValid } from '@hapi/address'
+import { isDisposableEmail } from 'disposable-email-domains-js'
+import { z } from 'zod'
+
+export const emailSchema = z
+  .string()
+  .email()
+  // @NOTE using @hapi/address here, in addition to the email() check to ensure
```

Comment:
Is there a reason not to only use @hapi/address?

Reply:
Internally, `zod` uses a regexp for validation. This validation strategy *could* be less permissive in some (edge) cases than `@hapi/address` as the latter uses an algorithm based on the spec. Truth is, I don't really know if the set of emails allowed by `@hapi/address` is covered by the set of emails allowed by `zod`, nor do I think we should investigate it.

Because of this uncertainty, and because other part of our codebases rely solely on `zod`, this code only allows emails that are valid according to both libraries ensuring that we never encounter a case where an email allowed here is in a format that would be rejected by other parts of our systems.

Reply:
Note that this behavior was not changed with this PR (code was moved)